aether-kernel/aether/kernel/api/models.py

```diff
@@ -160,6 +208,28 @@ class Meta:
         ]
 
 
+class SubmissionMapping(ExportModelOperationsMixin('kernel_submissionmapping'), TimeStampedModel):
```

Comment:
Maybe I don't see the whole picture, do we need this model? and why?

Reply:
This is used to keep track of the actual mappings that were used during entity extraction for replay purposes. (mapping revisions included). Bear in mind that mappings can be added or removed to a set after a submission has been made.

Reply:
But that should be reflected in the Entity and not in the Submission. The Submission must be "agnostic" and "ignorant" of all the triggered actions. Or am I wrong?

Reply:
I tend to agree with @obdulia-losantos here, but I'm not sure - what do you think @quesebifurcan ?

Reply:
Yes, agreed, the submission is akin to "raw" data and shouldn't (need) to store too much other information. I have no idea how replay will actually work, but for the most basic version, the submission would only need to know which mapping set it was submitted against. This is the way it currently works, at least.

Reply:
Knowing the mapping set a submission was made against doesn't mean it knows the actual mapping(s) that was used during the extraction
I see the point of having the mapping refs on the entity, which currently is implicitly referenced through the submission. I am assuming that is why the `map_revision` is on the submission. But with the introduction of mapping sets that will not be the case because the mappings in a mapping set are mutable.
For a basic version without considering replay, i agree, submission should be ignorant. It is interesting though to know why `map_revision` is currently in the submission.